### PR TITLE
RTL and localized messages support

### PR DESCRIPTION
**Pull request by CodeBaboon**
Added some styling for RTL languages, only applies to the generic plugin at this point.
Brought in react-frau-intl to handle localization of messages, specifically the word
'download' that appears in the generic viewer.

### CODE REVIEW

File: src/plugins/generic/generic.scss [hunk 1]
```diff
@@ -14,12 +14,19 @@
...
+		[dir="rtl"] & {
+			text-align: right;
+		}
 	}
 
 	.vui-fileviewer-icon {
 		float: left;
 		margin: 15px 40px 0 20px;
 		vertical-align: middle;
+
```
Comment (dlockhart) [inline src/plugins/generic/generic.scss]: You'll want to flip the margins around here too likely.

File: package.json [hunk 2]
```diff
@@ -31,10 +31,12 @@
   "dependencies": {
     "bower": "^1.6.5",
     "d2l-intl": "^0.2.1",
+    "react-frau-intl": "^0.1.2",
     "reactify": "^1.1.1"
   },
   "devDependencies": {
     "babel-jest": "^5.3.0",
+    "babelify": "^6.1.2",
```
Comment (dlockhart) [inline package.json]: So in theory, `react-frau-intl` should cause babelify to be included and run since it defines it in its own `package.json`. i.e. you shouldn't need to know its internals and that it's using ES6. If that's not happening, we should fix it.

File: src/lang/en.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+	"Plugins": {
+		"Generic": {
+			"Download": "Download"
+		}
+	}
+}
```
Comment (dlockhart) [inline src/lang/en.json]: If this is the only language term the component needs (unlikely I guess), you can likely find it and all its translations in the monolith that you could copy in here.

File: package.json (hunk 2 above)
Comment (CodeBaboon) [inline package.json]: That's definitely not happening. This small change actually caused me a lot of problems because of the ES6. I couldn't compile without Babelify, so then I went and sucked it down but unknowingly ended up with the newer version (7+) which uses Babel 6 and all the presets and whatnot. A bunch of confused attempts later I finally realised what was going on but still failed to get that version of Babelify to work before falling back to v6 and getting it going.

Long story short if we shouldn't need babelify that would be awesome to fix for anyone else trying to consume the `react-frau-intl` :)

Comment (omsmith) [inline package.json]: babelify need to be added to "dependencies" in react-frau-ibtl. That's all.

File: package.json [hunk 4]
```diff
@@ -31,10 +31,12 @@
   "dependencies": {
     "bower": "^1.6.5",
     "d2l-intl": "^0.2.1",
+    "react-frau-intl": "^0.1.2",
     "reactify": "^1.1.1"
```
Comment (dbatiste) [inline package.json]: Why is reactify a dependency instead of a devDependency?  

Aside, babelify provides excellent support for JSX as well, perhaps even better than reactify now.  It most certainly provides better ES6/ES7 transpiling support, but that may not be a consideration for your app at this time.  Either way, babelify appears to be a lot more active, and I suspect it's JSX support will be better as changes are made to React.

Comment (CodeBaboon) [inline package.json]: I'm not sure why it is in dep and not devDep. After this PR I think I'll try again moving to Babelify 7+, I suspect reactify might have been part of my problem and I also have a couple more ideas to get past my earlier issues. Regardless, if I can pull it together in not too much time I'll whip up another PR that moves us up and removes reactify etc.

File: package.json (hunk 2 above)
Comment (CodeBaboon) [inline package.json]: https://github.com/Brightspace/react-frau-intl/pull/17 should do it, then when we consume `react-frau-intl` we don't even need to concern ourself with babelify at all. If/when `react-frau-intl` merges that and version bumps then we can update to it and remove the babelify lines from this package.json.

**Comment by CodeBaboon** [comment]
@omsmith @dlockhart Assuming Travis is cool with the react-frau-intl update are we good for me to merge this guy?

File: src/plugins/generic/generic.scss (hunk 1 above)
Comment (dlockhart) [inline src/plugins/generic/generic.scss]: Nitpick: you could just override the left/right margins here so you don't have to double-specify the top/bottom in 2 places.

**Comment by dlockhart** [comment]
Yeah looks good. Let's chat separately (different PR) about how we could handle requests for locales we could support (like a different dialect of French, Spanish, etc.) but that might not map to one of the languages you have here. [i10n-language-builder](https://github.com/Brightspace/l10n-language-builder) might be able to help.

**Comment by CodeBaboon** [comment]
Ok, thanks Dave. Once Travis get its act together I'll merge this thing and we can talk locales/langs etc separately.

File: sample/app.js [hunk 5]
```diff
@@ -30,7 +30,11 @@ var Main = React.createClass({
 		this.setState({file: files[event.target.selectedIndex - 1]});
 	},
 	localeSelected: function(event) {
-		this.setState({locale: event.target.value});
+		var dir = event.target.value === 'ar-SA' ? 'rtl' : 'ltr';
```
Comment (jstefaniuk-d2l) [inline sample/app.js]: I think this also may need to look for 'ar':
https://github.com/Brightspace/intl/blob/master/src/locale-data/ar.json

File: sample/app.js [hunk 6]
```diff
@@ -57,10 +61,20 @@ var Main = React.createClass({
...
+						<option value="">-- Select a Locale --</option>
+						<option value="ar-SA">AR Arabic (ar-SA)</option>
+						<option value="en-CA">CDN English(en-CA)</option>
+						<option value="en-CA">UK English(en-GB)</option>
+						<option value="en-US">US English(en-US)</option>
+						<option value="es-MX">ES Spanish(es-MX)</option>
+						<option value="fr-CA">Français Canada(fr-CA)</option>
+						<option value="en-US">JA Japanese(ja)</option>
+						<option value="ko-KR">KR Korean (ko-KR)</option>
+						<option value="en-US">Português Brasil(pr-BR))</option>
```
Comment (jstefaniuk-d2l) [inline sample/app.js]: did you mean to use 'en-US' as the value for the last 5 options?

Comment (jstefaniuk-d2l) [inline sample/app.js]: And I think this line is pt-BR instead of pr-BR

File: src/plugins/generic/download.js [hunk 7]
```diff
@@ -3,6 +3,10 @@
 var React = require('react');
 
 var Download = React.createClass({
+	contextTypes : {
+		getIntlMessage: React.PropTypes.func,
+		formatMessage: React.PropTypes.func
```
Comment (jstefaniuk-d2l) [inline src/plugins/generic/download.js]: I don't think you need the formatMessage context here

File: src/getMessages.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var languages = {
```
Comment (jstefaniuk-d2l) [inline src/getMessages.js]: I notice that there are more language packs in the intl package (Such as 'en' and 'ar'): https://github.com/Brightspace/intl/tree/master/src/locale-data, should these also exist here, or will we not come upon those language strings? @dlockhart

Comment (dlockhart) [inline src/getMessages.js]: They should yes, which is why I mentioned the other day we could chat about it later. We built [i10n-language-builder](https://github.com/Brightspace/l10n-language-builder) to help with this. It lets you just define the base language files (e.g. en.json, ar.json, etc.) and any overrides (if necessary) and then will generate ALL the combinations for you as part of your build.

File: sample/app.js (hunk 6 above)
Comment (CodeBaboon) [inline sample/app.js]: Good catches, fix coming.

File: src/getMessages.js (hunk 8 above)
Comment (CodeBaboon) [inline src/getMessages.js]: Awesome stuff, we can look into that as a separate PR.
